File: fanpage-backend/src/main/java/com/example/fanpagebackend/modules/friend/service/FriendService.java
package com.example.fanpagebackend.modules.friend.service;

import com.example.fanpagebackend.modules.friend.dto.response.FriendRequestResponse;
import com.example.fanpagebackend.modules.user.dto.response.UserSummaryResponse;

import java.util.List;

public interface FriendService {
    FriendRequestResponse sendRequest(Long userId);
    FriendRequestResponse acceptRequest(Long requestId);
    void rejectRequest(Long requestId);
    void removeFriend(Long userId);
    List<UserSummaryResponse> getFriends();
    List<FriendRequestResponse> getReceivedRequests();
    List<FriendRequestResponse> getSentRequests();
    String getFriendshipStatus(Long otherUserId);
}

File: fanpage-frontend/src/services/friendService.js
import apiClient from './apiClient';

const normalizeUser = (user) => ({
  ...user,
  avatar: user?.avatar || null,
  name: user?.name || user?.fullName || 'Người dùng',
  bio: user?.bio || '',
  friendshipStatus: user?.friendshipStatus || 'NONE'
});

const normalizeRequest = (request) => ({
  ...request,
  requester: normalizeUser(request.requester),
  addressee: normalizeUser(request.addressee)
});

const friendService = {
  async getFriends() {
    const response = await apiClient.get('/friends');
    return (response.data || []).map(normalizeUser);
  },

  async getReceivedRequests() {
    const response = await apiClient.get('/friends/requests/received');
    return (response.data || []).map(normalizeRequest);
  },

  async getSentRequests() {
    const response = await apiClient.get('/friends/requests/sent');
    return (response.data || []).map(normalizeRequest);
  },

  async sendRequest(userId) {
    const response = await apiClient.post(`/friends/requests/${userId}`, {});
    return normalizeRequest(response.data);
  },

  async acceptRequest(requestId) {
    const response = await apiClient.post(`/friends/requests/${requestId}/accept`, {});
    return normalizeRequest(response.data);
  },

  async rejectRequest(requestId) {
    return apiClient.post(`/friends/requests/${requestId}/reject`, {});
  },

  async removeFriend(userId) {
    return apiClient.delete(`/friends/${userId}`);
  }
};

export default friendService;
